lib/project_types/script/layers/application/push_script.rb

```diff
@@ -6,18 +6,16 @@ module Application
       class PushScript
         class << self
           def call(ctx:, language:, extension_point_type:, script_name:, api_key:, force:)
-            extension_point = ExtensionPoints.get(type: extension_point_type)
             script = Infrastructure::ScriptRepository.new.get_script(language, extension_point_type, script_name)
-            ProjectDependencies
-              .install(ctx: ctx, language: language, extension_point: extension_point, script_name: script_name)
+            ProjectDependencies.install(ctx: ctx, language: language)
             BuildScript.call(ctx: ctx, script: script)
             push_script(ctx, script, api_key, force)
           end
 
           private
 
           def push_script(ctx, script, api_key, force)
-            compiled_type = Infrastructure::ScriptBuilder.for(script).compiled_type
+            compiled_type = Infrastructure::TaskRunner.for(ctx, script.language).compiled_type
```

Comment:
I really think we should get the `TaskRunner` class type once, and inject it into other places such as `ProjectDependencies` and `BuildScript` classes that uses it, just to be consistent.  Otherwise, I can see there is room for error if each of these classes determines the correct `TaskRunner` type to use.  So, it can be just:
```
    task_runner = Infrastructure::TaskRunner.for(ctx, script.language)
    ProjectDependecies.install(task_runner, ctx, extension_point, script_name)
```